File: testSuite/test-impulsiveHeating.pl
#!/usr/bin/env perl
use strict;
use warnings;
use PDL;
use PDL::NiceSlice;
use PDL::IO::HDF5;
use Data::Dumper;

# Run models that test that the impulsive heating model works as expected.
# Andrew Benson (02-June-2022)

# Make output directory.
system("mkdir -p outputs/");

# Run the impulsive heating model.
system("export OMP_NUM_THREADS=1; cd ..; ./Galacticus.exe testSuite/parameters/impulsiveHeating.xml");
if ( $? == 0 ) {
    print "success: impulsive heating model ran successfully\n";
} else {
    print "FAIL: impulsive heating model failed to run\n";
}

exit;

File: testSuite/test-impulsiveHeating.py
#!/usr/bin/env python3
import subprocess
import sys

# Test impulsive heating model.
# Andrew Benson (ported to Python)

# Ensure output directory exists.
subprocess.run("mkdir -p outputs", shell=True)

# Run the model.
status = subprocess.run("export OMP_NUM_THREADS=1; cd ..; ./Galacticus.exe testSuite/parameters/impulsiveHeating.xml", shell=True)
if status.returncode == 0:
    print("success: impulsive heating model ran successfully")
else:
    print("FAIL: impulsive heating model failed to run")
